### PR TITLE
Move adapter into subnamespace

### PR DESCRIPTION
**Pull request by ddeboer**
As decided in https://github.com/php-http/httplug/issues/98.

### CODE REVIEW

File: tests/HttpAdapterTest.php [hunk 1]
```diff
@@ -1,22 +1,22 @@
 <?php
 
-namespace Http\Adapter\Tests;
+namespace Http\Adapter\Guzzle6\Tests;
 
 use GuzzleHttp\Client;
-use Http\Adapter\Guzzle6HttpAdapter;
+use Http\Adapter\Guzzle6\Client as Adapter;
```
Comment (ddeboer) [inline tests/HttpAdapterTest.php]: This is kind of weird. Use the `Client` class as an `Adapter` that actually wraps the Guzzle `Client`. I still prefer `Http\Adapter\Guzzle6\Client` over `Http\Adapter\Guzzle6\Adapter` though, as it makes clear that this class implements the `HttpClient` interface.

Comment (sagikazarmark) [inline tests/HttpAdapterTest.php]: I think this is something that should be discussed generally, as it affects the whole Adapter naming idea. Can you open an issue in httplug to discuss this?

Comment (ddeboer) [inline tests/HttpAdapterTest.php]: See https://github.com/php-http/httplug/issues/98#issuecomment-167771941.

Comment (ddeboer) [inline tests/HttpAdapterTest.php]: Hm, let’s not forget that `Http\Adapter\Guzzle6\Adapter` looks very much redundant. There’s still the option `Http\Adapter\Guzzle6\Guzzle6Adapter|Client` of course.

Comment (sagikazarmark) [inline tests/HttpAdapterTest.php]: How about `Http\Guzzle6Adapter` namespace?

Comment (ddeboer) [inline tests/HttpAdapterTest.php]: Would be fine with me as well.

Comment (sagikazarmark) [inline tests/HttpAdapterTest.php]: `Http\Guzzle6Adapter\Client` would be fine by me. It reflects that we are talking about an adapter, but the object itself can be called client.

Comment (sagikazarmark) [inline tests/HttpAdapterTest.php]: Can you please update the PR then?

Comment (ddeboer) [inline tests/HttpAdapterTest.php]: Will do, but still waiting for a definite decision in https://github.com/php-http/httplug/issues/98.

Comment (ddeboer) [inline tests/HttpAdapterTest.php]: Okay, if we stick with `Http\Adapter\Guzzle6\Client` it seems this PR doesn’t need to be changed and is ready for merging.

File: tests/HttpAdapterTest.php [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
-abstract class Guzzle6HttpAdapterTest extends HttpClientTest
+abstract class HttpAdapterTest extends HttpClientTest
 {
     /**
      * {@inheritdoc}
      */
     protected function createHttpAdapter()
     {
-        return new Guzzle6HttpAdapter(new Client(['handler' => $this->createHandler()]));
+        return new Adapter(new Client(['handler' => $this->createHandler()]));
```
Comment (sagikazarmark) [inline tests/HttpAdapterTest.php]: It is only test, but it is a bit confusing. Should we change this to Client and GuzzleClient?

Comment (ddeboer) [inline tests/HttpAdapterTest.php]: Done.

**Comment by ddeboer** [comment]
Updated the tests, so ready for merging.

**Comment by sagikazarmark** [comment]
Cool, great job @ddeboer

**Pull request merged**